urlutil.py

```diff
@@ -26,3 +31,18 @@ def get_local_file_from_url(url, tempdir):
             output.write(stream.read())
 
     return pathname
+
+def find_firmware_by_hash(hash, board, language):
```

Comment:
We're mixing "URL things" with "CircuitPython firmware things" here -- maybe this should be in a different util class.

Reply:
moved to `s3util` in latest commit, lmk if you prefer different name.